File: Server (ASP.NET Core)/Petadmin.Core/Models/Debt.cs
using System;
using PetAdmin.Core.Interfaces;

namespace Petadmin.Core.Models
{
    public class Debt : IGenericDbEntity
    {
        /// <summary>
        /// Keeps the database id of the debt.
        /// </summary>
        public int DebtId { get; set; }

        /// <summary>
        /// Keeps the database id of the debt owner.
        /// </summary>
        public int OwnerId { get; set; }

        /// <summary>
        /// Gets or sets the date of debt.
        /// </summary>
        public DateTime? DebtDate { get; set; }

        /// <summary>
        /// Gets or sets the name of the animal.
        /// </summary>
        public string AnimalName { get; set; }

        /// <summary>
        /// Gets or sets the cause of the debt.
        /// </summary>
        public string Cause { get; set; }

        /// <summary>
        /// Gets or sets the amount of the debt.
        /// </summary>
        public int DebtAmount { get; set; }

        /// <summary>
        /// Gets or sets the paid amount of the debt.
        /// </summary>
        public int PaidAmount { get; set; }
    }
}

File: Angular Client/models/debt.ts
import {FormGroup, Validators} from '@angular/forms';

export interface Debt {
  ownerId: number;
  debtId: number;
  debtDate: Date;
  animalName: string;
  cause: string;
  debtAmount: number;
  paidAmount: number;
}

export class DebtEntity implements Debt{
  constructor() {
    this.ownerId = 0;
    this.debtId = 0;
    this.debtDate = new Date();
    this.animalName = '';
    this.cause = '';
    this.debtAmount = 0;
    this.paidAmount = 0;
  }

  ownerId: number;
  debtId: number;
  debtDate: Date;
  animalName: string;
  cause: string;
  debtAmount: number;
  paidAmount: number;

  setData(form: FormGroup): void{
    this.ownerId = form.get('ownerId')?.value;
    this.debtId = form.get('debtId')?.value;
    this.debtDate = form.get('debtDate')?.value;
    this.animalName = form.get('animalName')?.value;
    this.cause = form.get('causeOfDebt')?.value;
    this.debtAmount = form.get('debtAmount')?.value;
    this.paidAmount = form.get('debtPaid')?.value;
  }
}
